src/modules/search/components/OmniSearch.tsx

```diff
@@ -0,0 +1,136 @@
+import SearchIcon from '@mui/icons-material/Search';
+import {
+  useAutocomplete,
+  Popper,
+  Box,
+  Paper,
+  Grid,
+  Typography,
+  AutocompleteGroupedOption,
+  MenuItem,
+  CircularProgress,
+} from '@mui/material';
+import { flatten, isEmpty } from 'lodash-es';
+import React, { useState } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+
+import TextInput from '@/components/elements/input/TextInput';
+import { useOmniSearchClientsQuery } from '@/types/gqlTypes';
+
+const OmniSearch: React.FC = () => {
+  const [value, setValue] = useState<string | null>(null);
+
+  const { data, loading } = useOmniSearchClientsQuery({
```

Comment:
Ideally the results would be sorted by relevancy, not last name. Probably a separate ticket, but let's look at what that would take to add that to the `ClientSearch` concern

Reply:
@gigxz Taking a cursory look at this, it seems like we could more easily do this if we were actually using the postgres full-text search mechanisms, which would be able to give us a clear ranking. Because it looks like we're currently doing a multi-condition search, we would probably have to roll our own ranking logic based on conditions matched, which honestly sounds like a nightmare.

One option to solve this could be to pre-generate the text search vectors for a given client, persist that information into the clients table (or some other place), and then search on it whenever we need to do text search ([source](https://www.compose.com/articles/mastering-postgresql-tools-full-text-search-and-phrase-search/)). That would mean adding some mechanism that would assemble the correct search vectors and handle maintaining that persisted information, and then refactoring the client search to use the text vectors instead of the multi-condition search it uses now. Alternatively we could also generate the vectors on every search (we do this in HNMI), which would trade the denormalization maintenance cost for a performance cost per query.

I'm not deeply experienced in postgres text searching–the above option came up from some googling and some experience in HNMI–so I don't know what the lift would be, or if it's even feasible. Definitely worth a new issue to handle that work, I think it's outside the scope here.

Reply:
Interesting, thanks for the research. Added this info to https://www.pivotaltracker.com/story/show/184065163